lib/communicator_stack.cpp

```diff
@@ -167,13 +167,17 @@ namespace quda
                                                                              stride);
   }
 
-  void comm_free(MsgHandle *&mh) { get_current_communicator().comm_free(mh); }
+#define CHECK_MH(mh) { if (mh == nullptr) errorQuda("null message handle"); }
```

Comment:
Can we make this a function instead of a macro?

Reply:
I had this as a macro so that the correct line and function location will be reported.  I think we should leave this as a macro until we can jump to C++20 (`std::source_location`)